cpp/arcticdb/storage/lmdb/lmdb_mock_client.hpp

```diff
@@ -64,7 +64,7 @@ class MockLmdbClient : public LmdbClientWrapper {
             KeyType key_type) const override;
 
 private:
-    std::map<LmdbKey, Segment> lmdb_contents;
+    std::map<LmdbKey, Segment> lmdb_contents_;
```

Comment:
Why is `std::map` used? Do we need the ordering it provides? If not `std::unordered_map` is a better choice.

Reply:
This was also discussed in a previous [PR comment](https://github.com/man-group/ArcticDB/pull/1395#discussion_r1516272965). In short, the `std::unordered_map` requires us to define `operator==` and a hash function for `LmdbKey` without bringing clear advantages.

Reply:
You can easily implement both operator== and the hash::function. There are a few options, you can inherit from std::pair<string, string>, you can keep the two fields inside as as pair (and provide accessor functions with meaningful names) or just convert to std::pair inside the hash function and operator== (similar to how you provide operator< for the unordered map).

Reply:
Yeah, let's do it

Reply:
Fixed in [4fab7ab](https://github.com/man-group/ArcticDB/pull/1414/commits/4fab7abb2dc80707a1c79363e8dea640186b35bd)